### DISCUSSION TITLE
CLI: New dotenv output format

### DISCUSSION THREAD

**Discussion by cimnine** [GitTools/GitVersion#4173] [Ideas]
I would appreciate it if the GitVersion CLI could output all the variables in `dotenv` format:

```sh
$ dotnet gitversion -output dotenv
AssemblySemFileVer=1.0.1.0
AssemblySemVer=1.0.1.0
BranchName=main
BuildMetaData=
CommitDate=2024-08-21
CommitsSinceVersionSource=0
…
```

This would allow more simplified CI/CD jobs with the two most important CI systems, GitLab CI and GitHub Actions. Both allow to pass environment variables to other jobs by writing them to a specific file (as documented for [GitLab](https://docs.gitlab.com/ee/ci/variables/#pass-an-environment-variable-to-another-job) and [GitHub](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#passing-values-between-steps-and-jobs-in-a-workflow)). The first step of the pipeline would calculate the GitVersion-variables and export them, and all subsequent jobs could use those variables.

One important thing to consider is what to do with `null` values. In the example above, `BuildMetaData` is `null`. If it's in the output, the environment variable will be defined as empty. But most probably, all GitVersion-variables that are `null` should not be printed at all in `dotenv` mode.

**Comment by arturcic** [comment]
I think it's a nice suggestion, so I will create an issue out of it
